Build fails for `aarch64-apple-watchos` target

Original [issue](https://github.com/anza-xyz/solana-sdk/issues/319).

Trying to build for watchOS but fails. Other Apple platforms succeeded (iOS, macOS, visionOS.,tvOS).

```bash
setoelkahfi@Heighliner1 wallet-kit-v3 % cargo +nightly build -Z build-std --target aarch64-apple-watchos
   Compiling quinn-udp v0.5.14
   Compiling indicatif v0.18.0
   Compiling governor v0.6.3
   Compiling anstyle-parse v0.2.7
   Compiling proc-macro2 v1.0.101
   Compiling async-channel v1.9.0
   Compiling ecdsa v0.16.9
   Compiling cargo_metadata v0.15.4
   Compiling solana-tls-utils v3.0.2
   Compiling fs-err v2.11.0
error[E0412]: cannot find type `mmsghdr` in crate `libc`
   --> /Users/setoelkahfi/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/quinn-udp-0.5.14/src/unix.rs:468:50
    |
468 |       let mut hdrs = unsafe { mem::zeroed::<[libc::mmsghdr; BATCH_SIZE]>() };
    |                                                    ^^^^^^^ help: a struct with a similar name exists: `cmsghdr`
    |
   ::: /Users/setoelkahfi/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/libc-0.2.175/src/unix/bsd/mod.rs:12:1
    |
 12 | / s! {
 13 | |     pub struct sockaddr {
 14 | |         pub sa_len: u8,
 15 | |         pub sa_family: sa_family_t,
...   |
134 | | }
    | |_- similarly named struct `cmsghdr` defined here

error[E0425]: cannot find function `recvmmsg` in crate `libc`
   --> /Users/setoelkahfi/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/quinn-udp-0.5.14/src/unix.rs:480:19
    |
480 |             libc::recvmmsg(
    |                   ^^^^^^^^ help: a function with a similar name exists: `recvmsg`
    |
   ::: /Users/setoelkahfi/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/libc-0.2.175/src/unix/bsd/mod.rs:782:5
    |
782 |     pub fn recvmsg(fd: c_int, msg: *mut crate::msghdr, flags: c_int) -> ssize_t;
    |     ---------------------------------------------------------------------------- similarly named function `recvmsg` defined here

Some errors have detailed explanations: E0412, E0425.
For more information about an error, try `rustc --explain E0412`.
error: could not compile `quinn-udp` (lib) due to 2 previous errors
warning: build failed, waiting for other jobs to finish...
```

--- a/quinn-udp/build.rs
+++ b/quinn-udp/build.rs
@@ -9,7 +9,8 @@ fn main() {
                 target_os = "macos",
                 target_os = "ios",
                 target_os = "tvos",
-                target_os = "visionos"
+                target_os = "visionos",
+                target_os = "watchos"
             )
         },
         bsd: {

